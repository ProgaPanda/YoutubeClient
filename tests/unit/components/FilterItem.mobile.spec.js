import { shallowMount } from '@vue/test-utils';
import FilterItemMobile from '@/components/FilterItem.mobile.component.vue';

describe('FilterItemMobile', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(FilterItemMobile, {
      propsData: {
        name: 'Type',
        options: ['All', 'Video', 'Channel', 'Playlist'],
        defaultOption: 'All',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders correct number of filter options', () => {
    const wrapper = shallowMount(FilterItemMobile, {
      propsData: {
        name: 'Type',
        options: ['All', 'Video', 'Channel', 'Playlist'],
        defaultOption: 'All',
      },
    });

    expect(wrapper.find('.filter-item__select').findAll('option')).toHaveLength(4);
  });
});
