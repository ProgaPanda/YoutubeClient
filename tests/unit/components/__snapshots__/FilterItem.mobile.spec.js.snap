// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItemMobile renders correctly 1`] = `
<div
  class="filter-item"
>
  <select
    class="filter-item__select"
  >
    <option>
      
      All
    
    </option>
    <option>
      
      Video
    
    </option>
    <option>
      
      Channel
    
    </option>
    <option>
      
      Playlist
    
    </option>
  </select>
</div>
`;
